modules/calc/src/main/java/com/opengamma/strata/calc/runner/CalculationParameter.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.calc.runner;
+
+import com.opengamma.strata.basics.CalculationTarget;
+import com.opengamma.strata.calc.config.Measure;
+
+/**
+ * The base interface for calculation parameters.
+ * <p>
+ * Parameters are used to control the calculation.
+ */
+public interface CalculationParameter {
+
+  /**
+   * Gets the type that the parameter will be queried by.
+   * <p>
+   * Parameters can be queried using {@link CalculationParameters#findParameter(Class)}.
+   * This type is the key that callers must use in that method.
+   * 
+   * @return the type of the parameter implementation
+   */
+  public abstract Class<?> queryType();
```

Comment:
I find this slightly puzzling. What is the query type? Is it expected to be the type implementing `CalculationParameter`? Can we do better with the type signature so that's clear. When I look at this interface in isolation the query type looks like a type that's floating around unconnected to anything.